apps/guardian-ui/src/components/setup/RunDKG.tsx

```diff
@@ -25,10 +25,18 @@ export const RunDKG: React.FC<Props> = ({ next }) => {
   const theme = useTheme();
   const [isWaitingForOthers, setIsWaitingForOthers] = useState(false);
   const [error, setError] = useState<string>();
+  const [ellipsis, setEllipsis] = useState('');
 
   // Poll for peers and configGenParams while on this page.
   useConsensusPolling();
 
+  useEffect(() => {
+    const interval = setInterval(() => {
+      setEllipsis((prev) => (prev.length < 3 ? prev + '.' : ''));
+    }, 500);
+    return () => clearInterval(interval);
+  }, []);
```

Comment:
Could we pull this `useEffect()` and the ellipsis `useState()` into a new hook? Claude generated this for me:

```
import React, { useState, useEffect } from 'react';

const useLoadingDots = (interval = 500) => {
  const [dots, setDots] = useState('');

  useEffect(() => {
    const timer = setInterval(() => {
      setDots(prevDots => {
        switch (prevDots) {
          case '':
            return '.';
          case '.':
            return '..';
          case '..':
            return '...';
          default:
            return '';
        }
      });
    }, interval);

    return () => clearInterval(timer);
  }, [interval]);

  return dots;
};

const LoadingText = ({ text, interval }) => {
  const dots = useLoadingDots(interval);
  return <div>{text}{dots}</div>;
};

export default function App() {
  return (
    <div className="flex items-center justify-center h-screen">
      <LoadingText text="Loading" interval={300} />
    </div>
  );
}
```

See the demo [here](https://claude.site/artifacts/cad51d0a-0c48-4a61-84e9-a470a79044e0).

Your code in the `useEffect()` is much more compact than the `switch` statement, but I think a separate hook would be cleaner.

Reply:
done